flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReaderTest.java

```diff
@@ -136,7 +156,67 @@ public void testBinlogReadFailoverCrossTransaction() throws Exception {
                     "+U[103, user_3, Shanghai, 123567891234]"
                 };
         // the 4 records are produced by 3 operations
-        List<String> restRecords = consumeRecords(restartReader, dataType, 3);
+        List<String> restRecords = consumeRecords(restartReader, dataType);
+        assertEqualsInOrder(Arrays.asList(expectedRestRecords), restRecords);
+        restartReader.close();
+    }
+
+    @Test
+    public void testBinlogReadFailoverCrossCheckpoint() throws Exception {
```

Comment:
The code in `testBinlogReadFailoverCrossTransaction` and `testBinlogReadFailoverCrossCheckpoint` is the same.